src/relay/pass/fold_conv2d.cc

```diff
@@ -0,0 +1,163 @@
+/*!
+ * Copyright (c) 2018 by Contributors
+ *
+ * \file fold_conv2d.cc
+ *
+ * \brief Fold multiple 2d convolutions into a single convolution.
+ *
+ * This pass replaces convolutions that share the same input node and the same arguments (except
+ * that the number of output channels can be different) with a single convolution. The weight of
+ * the new 2d convolution is the concatenation of the original weights.
+ *
+ * This prevents launching multiple kernels in networks with multiple convolution branches, such
+ * as Inception block.
+ */
+
+#include <tvm/relay/pass.h>
+#include <tvm/relay/expr_functor.h>
+#include <tvm/relay/attrs/nn.h>
+#include <tvm/relay/attrs/transform.h>
+#include "./expr_subst.h"
+#include "./pattern_util.h"
+
+namespace tvm {
+namespace relay {
+
+class SiblingConv2DFinder : public ExprVisitor {
+ public:
+  std::unordered_map<Expr, std::vector<const CallNode*>, NodeHash, NodeEqual> Find(
+      const Expr& expr) {
+    this->VisitExpr(expr);
+    return std::move(children_map_);
+  }
+
+  void VisitExpr_(const CallNode* n) final {
+    static const Op& conv2d = Op::Get("nn.conv2d");
+    ExprVisitor::VisitExpr_(n);
+    if (n->op.same_as(conv2d) && n->attrs.as<Conv2DAttrs>()->groups == 1) {
+      children_map_[n->args[0]].push_back(n);
+    }
+  }
+
+ private:
+  std::unordered_map<Expr, std::vector<const CallNode*>, NodeHash, NodeEqual> children_map_;
+};
+
+std::tuple<Expr, IndexExpr> TransformWeight(std::vector<const CallNode*> convolutions) {
+  int64_t num_filters = 0;  // number of filters of the transformed weight
+  Array<Expr> weights;
+  for (const CallNode* n : convolutions) {
+    weights.push_back(n->args[1]);
+    auto channels = as_const_int(n->attrs.as<Conv2DAttrs>()->channels);
+    CHECK(channels);
+    num_filters += *channels;
+  }
+  return std::tuple<Expr, IndexExpr>{MakeConcatenate(TupleNode::make(weights), 0),
```

Comment:
you can use std::make_tuple().
Or replace std::tuple with std::pair and use std::make_pair().